Add documentation for changes introduced by jslib-aws v0.7.0 (#939)
grafana/k6-docs: src/data/markdown/docs/20 jslib/01 jslib/01 aws/SignatureV4/02 presign().md

---
title: 'presign'
description: 'Signaturev4.presign pre-signs a URL with the AWS Signature V4 algorithm'
excerpt: 'SignatureV4.sign pre-signs a URL with the AWS Signature V4 algorithm'
---

> `SignatureV4.presign()` pre-signs a URL with the AWS Signature V4 algorithm. Given an HTTP request description, it returns a new HTTP request with the AWS signature v4 authorization added. It returns an Object holding a `url` containing the authorization information encoded in its query string, ready to use in the context of k6's HTTP call.

Reviewer: ```suggestion
`SignatureV4.presign()` pre-signs a URL with the AWS Signature V4 algorithm. Given an HTTP request description, it returns a new HTTP request with the AWS signature v4 authorization added. It returns an Object holding a `url` containing the authorization information encoded in its query string, ready to use in the context of a k6 HTTP call.
```

| uriEscapePath | boolean                  | Whether to uri-escape the request URI path as part of computing the canonical request string. This is required for every AWS service, except Amazon S3, as of late 2017.                                                                                                |
| applyChecksum | boolean                  | Whether to calculate a checksum of the request body and include it as either a request header (when signing) or as a query string parameter (when pre-signing). This is required for AWS Glacier and Amazon S3 and optional for every other AWS service as of late 2017. |
|               |                          |                                                                                                                                                                                                                                                                         |

> It is also possible to provide further options and override the SignatureV4's options in the context of this specific request. This is done by passing a second parameter to the `presign` method, which is an Object with the following parameters.

Reviewer: ```suggestion
You can provide further options and override SignatureV4 options in the context of this specific request.
To do this, pass a second parameter to the `presign` method, which is an Object with the following parameters.
```

| Property          | Type          | Description                                                                |
| :---------------- | :------------ | :------------------------------------------------------------------------- |

> | expiresIn         | number              | The number of seconds before the presigned URL expires                                                                            |

Author: Hehe I did a pass before pushing the PR, but it seems some went through still, will try to catch the rest indeed 🤞🏻 🙇🏻

| signingDate       | Date          | overrides the Date used in the signing operation                           |
| signingService    | string        | overrides the signer's AWS service in the context of the `sign` operation. |
| signingRegion     | string        | overrides the signer's AWS region in the context of the `sign` operation   |
| unsignableHeaders | `Set<string>` | excludes headers from the signing process                                  |
| signableHeaders   | `Set<string>` | mark headers as signed                                                     |

### Returns

The `presign` operation returns an Object with the following properties.

| Property | Type   | Description                                                                |
| :------- | :----- | :------------------------------------------------------------------------- |

> | headers  | Object | The pre-signed request's headers to use in the context of a k6 HTTP request |

Reviewer: ```suggestion
| headers  | Object | The pre-signed request headers to use in the context of a k6 HTTP request |
```
